src/vs/workbench/contrib/terminal/browser/terminalView.ts

```diff
@@ -405,12 +399,13 @@ class SingleTerminalTabActionViewItem extends MenuEntryActionViewItem {
 		this._register(this._terminalService.onDidChangeInstanceColor(e => this.updateLabel(e.instance)));
 		this._register(this._terminalService.onDidChangeInstanceTitle(e => {
 			if (e === this._terminalGroupService.activeInstance) {
-				this._action.tooltip = getSingleTabTooltip(e, this._terminalService.configHelper.config.tabs.separator);
+				// TODO: Fix types
+				(this._action as unknown as Action).tooltip = getSingleTabTooltip(e, this._terminalService.configHelper.config.tabs.separator);
```

Comment:
@Tyriar  This is the error I mentioned in the PR description. `MenuItemAction` says its properties are all readonly, but we then try writing them here

Reply:
It may be `readonly` on `MenuItemAction` but we're dealing with an `IAction` here (surprised TS never gave us an error for `MenuItemAction implements IAction`). Changing the tooltip is important here as the terminal title is dynamic, can you come up with a solution for this too so this doesn't just create another debt item that probably won't get looked at for a long time.